### PR TITLE
follow up getKernel version (e.g version 5.4.x.x.x)

### PR DESCRIPTION
**Pull request by jichenjc**
Signed-off-by: jichenjc <jichenjc@cn.ibm.com>

### CODE REVIEW

**Comment by jichenjc** [comment]
after change, it print 

config getKernelVersion:  5.4

File: pkg/config/config.go [hunk 1]
```diff
@@ -82,11 +82,19 @@ func getKernelVersion(c Client) float32 {
 	}
 	// per https://github.com/google/cadvisor/blob/master/machine/info.go#L164
 	kv := utsname.Release[:bytes.IndexByte(utsname.Release[:], 0)]
+	// for something like 4.18.10-24-generic
 	val, err := strconv.ParseFloat(string(kv[:4]), 32)
 	if err == nil {
 		return float32(val)
+	} else {
+		// for something like 5.4.0-48-generic
```
Comment (marceloamaral) [inline pkg/config/config.go:90]: `5.4.0-48-generic` and `4.18.10-24-generic` looks very similar to me
What is the problem here?

Comment (jichenjc) [reply pkg/config/config.go:90]: check https://github.com/sustainable-computing-io/kepler/issues/179#issuecomment-1253069991

Comment (rootfs) [reply pkg/config/config.go:90]: what about kernel `5.0, 6.0`?
https://en.wikipedia.org/wiki/Linux_kernel_version_history

how about splitting the string by `.`,  taking the first two substrings, and recreating a new kernel string?

File: pkg/config/config.go [hunk 2]
```diff
@@ -82,11 +82,19 @@ func getKernelVersion(c Client) float32 {
 	}
 	// per https://github.com/google/cadvisor/blob/master/machine/info.go#L164
 	kv := utsname.Release[:bytes.IndexByte(utsname.Release[:], 0)]
+	// for something like 4.18.10-24-generic
 	val, err := strconv.ParseFloat(string(kv[:4]), 32)
 	if err == nil {
 		return float32(val)
+	} else {
```
Comment (rootfs) [inline pkg/config/config.go:89]: how about splitting the string by `.`, and taking the first two substrings, and recreating the new kernel string?

**Comment by jichenjc** [comment]
@rootfs done

File: pkg/config/config_test.go [hunk 3]
```diff
@@ -70,11 +70,27 @@ var _ = Describe("Test Configuration", func() {
...
 		Expect(true).To(Equal(valid))
 
 		tempUnixName = "3.10"
-		valid = getKernelVersion(mockc) > cGroupIDMinKernelVersion
+		v = getKernelVersion(mockc)
+		valid = v > cGroupIDMinKernelVersion
+		Expect(-1).NotTo(Equal(v))
+		Expect(false).To(Equal(valid))
+
+		tempUnixName = "3.1"
```
Comment (rootfs) [inline pkg/config/config_test.go:90]: a bit paranoid but can you add a case `6.0-rc6`?

**Comment by rootfs** [comment]
thank you @jichenjc, one small suggestion, ready to go

File: pkg/config/config.go (hunk 2 above)
Comment (marceloamaral) [reply pkg/config/config.go]: +1

**Review by marceloamaral — APPROVED** [review]
Good to go now
/lgtm

**Pull request merged**